File: apps/todo_service/src/app/HTTP/http.php
<?php

use App\HTTP\Controllers\TodoController;
use App\Domain\Services\TodoService;
use App\Domain\Repositories\TodoRepository;
use App\HTTP\Middlewares\LoggerMiddleware;
use App\Lib\HTTPMethod;
use App\Lib\HTTPServer;

require __DIR__ . '/../../../vendor/autoload.php';
require __DIR__ . '../../../lib/database.php';

$todoController = new TodoController(
    todoService: new TodoService(
        todoRepository: new TodoRepository()
    )
);

$server = new HTTPServer();

$server->addMiddleware('/todo', LoggerMiddleware::class);

$server->addRoute(HTTPMethod::GET, '/todo', [$todoController, 'index']);
$server->addRoute(HTTPMethod::GET, '/todo/{todoId}', [$todoController, 'show']);
$server->addRoute(HTTPMethod::POST, '/todo', [$todoController, 'create']);
$server->addRoute(HTTPMethod::PUT, '/todo/{todoId}', [$todoController, 'update']);
$server->addRoute(HTTPMethod::DELETE, '/todo/{todoId}', [$todoController, 'destroy']);


$server->run();

File: apps/python_service/app/http/http.py
from app.http.controllers.todo_controller import TodoController
from app.http.http_server import HTTPServer


def start_http_server(domain):
    todo_controller = TodoController(domain.todo_service)

    http_server = HTTPServer()

    http_server.add_route('GET', '/todo', todo_controller.index)
    http_server.add_route('GET', '/todo/<todo_id>', todo_controller.show)
    http_server.add_route('POST', '/todo/', todo_controller.create)
    http_server.add_route('PUT', '/todo/<todo_id>', todo_controller.update)
    http_server.add_route('DELETE', '/todo/<todo_id>', todo_controller.destroy)

    http_server.run()
